dffml/source/hdfs_source.py

```diff
@@ -0,0 +1,58 @@
+import json
+from typing import AsyncIterator, NamedTuple, Dict
+from collections import OrderedDict
+import os
+from hdfs import *
+
+from dffml.base import BaseConfig
+from dffml.repo import Repo
+from dffml.source.source import BaseSourceContext, BaseSource
+from dffml.util.cli.arg import Arg
+from dffml.util.entrypoint import entry_point
+
+
+class HDFSSourceConfig(BaseConfig, NamedTuple):
+    host: str
+    port: str
+    user: str
+    filepath: str
+    source: BaseSource.load
+
+
+@entry_point("dffml.source")
+class HDFSSource(BaseSource):
+
+    CONTEXT = BaseSourceContext
+
+    async def __aenter__(self) -> "BaseSource":
+        self.client = InsecureClient(
+            "http://" + self.config.host + ":" + self.config.port,
+            user="hadoopuser",
+        )
```

Comment:
```python


async def new_open(source):
    with self.client.read(self.config.filepath, encoding="utf-8") as fd:
        await source.load_fd(fd)
self.config.source._open = new_open
```

Also, give it a new close method, which will be called when we exit from the filesource's context.

To enter the context, we usually do the `async with` statement:

```python
async with self.config.source as source: # this is where new_open get called
    # At the end of this with block new_close would get called, but we don't want to call it until the __aexit__ method of HDFSSource
```

You'll need to do something like this, also make sure to modify `__aexit__` similarly.

https://github.com/intel/dffml/blob/927fb736dd886b4976616f917633696dd3799e49/dffml/df/memory.py#L780-L792

You'll use

`self.source = await self.__stack.enter_async_context(self.config.source)`

Reply:
Implement __call__ method for HDFS source which does: `return self.source()`

Reply:
https://github.com/intel/dffml/blob/eb9b6002d6fb4df83f79dd57ae8992fb372413ad/dffml/source/source.py#L60-L61

Reply:
Done :+1: